src/libstd/path.rs

```diff
@@ -525,6 +525,26 @@ impl<'a> Hash for PrefixComponent<'a> {
 ///
 /// See the module documentation for an in-depth explanation of components and
 /// their role in the API.
+///
+/// This `struct` is created from iterating over the [`path::Components`]
```

Comment:
I'm not sure to understand in here. You're talking about the `Components` struct but this is supposed to be the doc for the `Component` enum. Any further explanation please? :)